packages/store/addon/-private/system/model/internal-model.ts

```diff
@@ -1229,9 +1243,13 @@ export default class InternalModel {
   }
 
   hasErrors() {
-    let errors = get(this.getRecord(), 'errors');
+    if (false) {
+      let errors = get(this.getRecord(), 'errors');
+      return errors.get('length') > 0;
+    } else {
+      return this._recordData.getErrors().length > 0;
```

Comment:
why would we assert for `isValid` but not for `hasErrors` ? I am now confused by the abandoned code branch above